core/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java

```diff
@@ -251,4 +259,97 @@ private void resolvePipelineAggregatorOrder(Map<String, AggregatorFactory> aggFa
             }
         }
     }
+
+    @Override
+    public AggregatorFactories readFrom(StreamInput in) throws IOException {
+        int factoriesSize = in.readVInt();
+        List<AggregatorFactory> factoriesList = new ArrayList<AggregatorFactory>(factoriesSize);
+        for (int i = 0; i < factoriesSize; i++) {
+            AggregatorFactory factory = in.readAggregatorFactory();
+            factoriesList.add(factory);
+        }
+        int pipelineFactoriesSize = in.readVInt();
+        List<PipelineAggregatorFactory> pipelineAggregatorFactoriesList = new ArrayList<PipelineAggregatorFactory>(pipelineFactoriesSize);
+        for (int i = 0; i < pipelineFactoriesSize; i++) {
+            PipelineAggregatorFactory factory = in.readPipelineAggregatorFactory();
+            pipelineAggregatorFactoriesList.add(factory);
+        }
+        AggregatorFactories aggregatorFactories = new AggregatorFactories(
```

Comment:
a bit weird to create an array list with a known size, just to convert it to an array. Let's build an array directly?